subsys/bluetooth/mesh/dfd_srv.c

```diff
@@ -346,10 +359,18 @@ static void upload_status_rsp(struct bt_mesh_dfd_srv *srv,
 		return;
 	}
 
-	net_buf_simple_add_u8(&rsp,
-			      bt_mesh_blob_srv_progress(&srv->upload.blob));
-	net_buf_simple_add_mem(&rsp, srv->upload.slot->fwid,
-			       srv->upload.slot->fwid_len);
+	if (srv->upload.is_oob) {
+#ifdef CONFIG_BT_MESH_DFD_SRV_OOB_UPLOAD
```

Comment:
This will keep the if-statement in the code, right? Maybe wrap the `if` into `ifdef` as well?

Reply:
I considered leaving the entire `is_oob` out when the KConfig option is disabled, but it turned out to be much simpler to just leave it in the code since it's used in a lot of the non-oob code as well. The reason I kept the `if` outside the `#ifdef` here is that it felt wrong to add the BLOB srv progress when `is_oob` is false, even though `is_oob` should never manage to become false with the KConfig option turned off. If `is_oob` somehow is true here with the KConfig option disabled, something has gone horribly wrong, and sending the status without progress seemed better than starting to call the blob server in that case.

Reply:
I never trust anything, as a principle. If someone writes a bug elsewhere, it should not propagate to a worse bug here. That was my idea, at least. If we are fine with just assuming there is no bug causing this to be set to the wrong value, then sure, we can compile it out.

Reply:
Looks like these are the only places where `is_oob` not under `ifdef`:
https://github.com/zephyrproject-rtos/zephyr/blob/517af9294e26dd2e994f4a0683f05458b603127d/subsys/bluetooth/mesh/dfd_srv.c#L462
https://github.com/zephyrproject-rtos/zephyr/blob/517af9294e26dd2e994f4a0683f05458b603127d/subsys/bluetooth/mesh/dfd_srv.c#L478
Not that much I think.

Reply:
So you are suggesting to remove it fully from the struct when the option is off? Or just move this `if` under the `ifdef` and keep those other ones in?

Reply:
I don't see point keeping it if we know it will never be changed.